File: src/main/java/com/intellij/tapestry/intellij/toolwindow/nodes/ResourceLeafNode.java
package com.intellij.tapestry.intellij.toolwindow.nodes;

import com.intellij.tapestry.core.resource.IResource;

import javax.swing.tree.DefaultMutableTreeNode;

/** A navigable leaf wrapping a single resource file (template, message catalog, ...). */
public class ResourceLeafNode extends DefaultMutableTreeNode {

    private final String _label;

    public ResourceLeafNode(IResource resource) {
        super(resource);
        // Precompute the label here (read action); toString() runs on the EDT during rendering.
        _label = resource.getName();
    }

    public String toString() {
        return _label;
    }
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/toolwindow/nodes/ResourceLeafNode.kt
package com.github.rar91279.plugin.tapestry.intellij.toolwindow.nodes

import com.intellij.psi.PsiFile
import javax.swing.tree.DefaultMutableTreeNode

/** A navigable leaf wrapping a single resource file (template, message catalog, ...). */
class ResourceLeafNode(resource: PsiFile) : DefaultMutableTreeNode(resource) {

    private val label = resource.name

    override fun toString() = label
}
